qemu/tests/migration_with_numa.py

```diff
@@ -0,0 +1,104 @@
+import re
+import logging
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context
+from virttest.utils_misc import get_mem_info
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Bind guest node0 and node1 to 2 host nodes, do migration test
+
+    1. Boot src guest with 2 numa node and all bind to 2 host numa nodes
+    2. Migration
+    3. Check the memory in guest
+    4. Check the memory in dest host
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def get_nodes_size(size_type='MemTotal', session=None):
+        """
+        Get the node size of each node in host/guest, sort with size.
+
+        :param size_type: the type of the node size
+        :param session: ShellSession object
+        """
+        numa_info = NumaInfo(session=session)
+        nodes_size = {}
+        numa_nodes = numa_info.online_nodes
+        for node in numa_nodes:
+            nodes_size[node] = float(numa_info.online_nodes_meminfo[node][size_type])
+        nodes_size = dict(sorted(nodes_size.items(), key=lambda item: item[1]))
+        return nodes_size
+
+    host_nodes_size = get_nodes_size(size_type='MemFree')
+    mem_devs = params.objects('mem_devs')
+    num = len(mem_devs)
+    for mem_dev in mem_devs:
+        size_mem = params.object_params(mem_dev).get('size_mem')
+        size_mem = float(normalize_data_size(size_mem, 'K'))
+        for node, size in host_nodes_size.items():
+            if int(size) >= int(size_mem):
+                params['host-nodes_mem_%s' % mem_dev] = str(node)
+                num -= 1
+                del host_nodes_size[node]
+                break
+
+    if num != 0:
+        test.cancel("No enough available nodes for testing!")
+
+    params['start_vm'] = 'yes'
+    env_process.preprocess(test, params, env)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    # do migration
+    mig_timeout = float(params.get("mig_timeout", "3600"))
+    mig_protocol = params.get("migration_protocol", "tcp")
+    vm.migrate(mig_timeout, mig_protocol, env=env)
+    session = vm.wait_for_login()
+
+    os_type = params["os_type"]
+    if os_type == 'linux':
+        error_context.context("Check the memory in guest", logging.info)
+        # Use 30 plus the gap of 'MemTotal' in OS and '-m' in cli as threshold
+        MemTotal = get_mem_info(session, 'MemTotal')
```

Comment:
We are not supposed to use capitalized variable names ?

Reply:
> We are not supposed to use capitalized variable names ?

`variable name` ?  It is a parameter value, the parameter name is 'attr' for function `get_mem_info`.
Do i misunderstanding ?

Reply:
I got your point, it is the first `MemTotal`, you are right, let me update it, thanks !

Reply:
Updated, could you help re-review, thanks !